### PR TITLE
Change morphing APIs to support LOD

### PR DESCRIPTION
**Pull request by daemyung**
@pixelflinger I have changed morphing APIs based on your comments. I would like this changes are what you want. I guess there is a lot of work for perfect implementation. But let's do it step by step.

### CODE REVIEW

**Review by pixelflinger — APPROVED** [review]
Sounds good.

**Review by romainguy — COMMENTED** [review]
The Java/JS APIs need to be updated as well

**Comment by daemyung** [comment]
@romainguy Please review Java and Web bindings. Actually I don't know JS, please let me know what should I do more if there are something to do.

**Comment by pixelflinger** [comment]
> @romainguy Please review Java and Web bindings. Actually I don't know JS, please let me know what should I do more if there are something to do.

I don't know js either... I just wing it :-)

**Review by prideout — CHANGES_REQUESTED** [review]
This JS binding will not work because there is no class binding for MorphTargetBuffer.  In fact the `setMorphWeights` method that was recently added won't work either (JS does not have `float*` so it should actually take emscripten::val or std::vector.)

I think it's better to omit bindings than to create bindings that won't work.  In this PR, maybe we could just remove `setMorphWeights` and create an issue to do this work later.

**Comment by pixelflinger** [comment]
looks like this need to be rebased/merged in top of #5079 ?

**Comment by daemyung** [comment]
> This JS binding will not work because there is no class binding for MorphTargetBuffer. In fact the `setMorphWeights` method that was recently added won't work either (JS does not have `float*` so it should actually take emscripten::val or std::vector.)

Thanks! I have updated it.

> I think it's better to omit bindings than to create bindings that won't work. In this PR, maybe we could just remove `setMorphWeights` and create an issue to do this work later.

Thanks for your advice. I remove this binding.

**Comment by daemyung** [comment]
> looks like this need to be rebased/merged in top of #5079 ?

I don't think so... is there any relations between two PRs?

**Review by prideout — APPROVED** [review]

**Comment by pixelflinger** [comment]
> > looks like this need to be rebased/merged in top of #5079 ?
> 
> I don't think so... is there any relations between two PRs?

Well, in this PR `MorphTargetBuffer.cpp ` is entirely new.

**Comment by daemyung** [comment]
> Well, in this PR `MorphTargetBuffer.cpp ` is entirely new.

@pixelflinger Yes because this is first time adding `MorphTargetBuffer``s Java binding.

**Review by bejado — APPROVED** [review]

**Review by pixelflinger — APPROVED** [review]

File: android/filament-android/src/main/java/com/google/android/filament/MorphTargetBuffer.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    /**
+     * Updates positions of morph target at the index.
+     *
+     * @param engine {@link Engine} instance
+     * @param targetIndex The index of morph target to be updated
+     * @param positions Pointer to at least count positions
+     * @param count Number of position elements in positions
+     */
+    public void setPositionsAt(@NonNull Engine engine,
+                               @IntRange(from = 0) int targetIndex,
```
[the remark this answers is not in the document]
Comment (daemyung) [reply android/filament-android/src/main/java/com/google/android/filament/MorphTargetBuffer.java:115]: Thanks. It's fixed.

**Pull request merged**